frontend/src/utilities/NavData.tsx

```diff
@@ -28,121 +29,140 @@ export const isNavDataHref = (navData: NavDataItem): navData is NavDataHref =>
 
-    if (operatorAvailable) {
-      navItems.push({
+  if (operatorAvailable) {
+    return [
+      {
         id: 'pipelines',
         group: { id: 'pipelines', title: 'Data Science Pipelines' },
         children: [
           { id: 'global-pipelines', label: 'Pipelines', href: '/pipelines' },
           { id: 'global-pipeline-runs', label: 'Runs', href: '/pipelineRuns' },
         ],
-      });
-    } else {
-      navItems.push({
-        id: 'pipelines',
-        label: (
-          <Split hasGutter>
-            <SplitItem>Data Science Pipelines</SplitItem>
-            <SplitItem>
-              <Icon status="danger" isInline>
-                <ExclamationCircleIcon />
-              </Icon>
-            </SplitItem>
-          </Split>
-        ),
-        href: `/dependency-missing/pipelines`,
-      });
-    }
+      },
+    ];
   }
 
-  if (featureFlagEnabled(dashboardConfig.spec.dashboardConfig.disableModelServing)) {
-    navItems.push({ id: 'modelServing', label: 'Model Serving', href: '/modelServing' });
-  }
+  return [
+    {
+      id: 'pipelines',
+      label: (
+        <Split hasGutter>
+          <SplitItem>Data Science Pipelines</SplitItem>
+          <SplitItem>
+            <Icon status="danger" isInline>
+              <ExclamationCircleIcon />
+            </Icon>
+          </SplitItem>
+        </Split>
+      ),
+      href: `/dependency-missing/pipelines`,
+    },
+  ];
+};
+
+const useModelServingNav = (): NavDataItem[] =>
+  useAreaCheck(SupportedArea.MODEL_SERVING, [
+    { id: 'modelServing', label: 'Model Serving', href: '/modelServing' },
+  ]);
+
+const useResourcesNav = (): NavDataHref[] => [
+  { id: 'resources', label: 'Resources', href: '/resources' },
+];
+
+const useCustomNotebooksNav = (): NavDataHref[] =>
+  useAreaCheck<NavDataHref>(SupportedArea.BYON, [
+    {
+      id: 'settings-notebook-images',
+      label: 'Notebook image settings',
+      href: '/notebookImages',
+    },
+  ]);
+
+const useClusterSettingsNav = (): NavDataHref[] =>
+  useAreaCheck<NavDataHref>(SupportedArea.CLUSTER_SETTINGS, [
+    {
+      id: 'settings-cluster-settings',
+      label: 'Cluster settings',
+      href: '/clusterSettings',
+    },
+  ]);
 
-  navItems.push({ id: 'resources', label: 'Resources', href: '/resources' });
+const useCustomRuntimesNav = (): NavDataHref[] =>
+  useAreaCheck<NavDataHref>(SupportedArea.CUSTOM_RUNTIMES, [
+    {
+      id: 'settings-custom-serving-runtimes',
+      label: 'Serving runtimes',
+      href: '/servingRuntimes',
+    },
+  ]);
 
-  const settingsNav = getSettingsNav(isAdmin, dashboardConfig);
-  if (settingsNav) {
-    navItems.push(settingsNav);
+const useUserManagementNav = (): NavDataHref[] =>
+  useAreaCheck<NavDataHref>(SupportedArea.USER_MANAGEMENT, [
+    {
+      id: 'settings-group-settings',
+      label: 'User management',
+      href: '/groupSettings',
+    },
+  ]);
+
+const useSettingsNav = (): NavDataGroup[] => {
+  // TODO: anyway to escape early?
```

Comment:
Remove TODO.
Not sure where you were trying to escape early but so long as you're using individual hooks, you cannot.
It would be possible to reimplement using the underlying `isAreaAvailable` util though.